text/3724-theupdateframework.md

```diff
@@ -0,0 +1,341 @@
+
+This proposal delegates the authority of selecting and managing the quorum membership to the Rust Project's Leadership Council. We recommend that the quorum be selected from trusted individuals within the Project and Foundation. This is a position of trust, not one of authority over anything other than safe handling of key material and flagging of unusual activity; the 9 members are responsible for executing quorum operations, and providing transparency and trust (through their refusal to participate in any malicious operations), but not for deciding independently what key operations should happen.
+
+These individuals should be available in the event quorum is needed for root key operations. These roles can and should be reappointed as needed, but to simplify logistics, these roles should not require rotation more often than 2-3 years. (Operations requiring quorum are expected to be rare, and an annual rotation would make rotation of the quorum group the vast majority of quorum operations.)
+
+##### When the Quorum will be needed
+[when-the-quorum-will-be-needed]: #when-the-quorum-will-be-needed
+
+- Changes to quorum membership or thresholds shall require a quorum (enforced by the HSM authentication)
+- Creation, signing, and issuance of new roles
+- Creation, signing, and modification of roles in the event of expiration or compromise
+
+##### Quorum Threat Mitigations
+
+- To reduce the feasibility of coercion or collusion, we recommend that no more than 3 (M-2) of the 9 people be affiliated with any one company or institution, or be residents or citizens of the same country. (If in the future the quorum threshold is changed, this limit should be changed accordingly to remain M-2.)
+- If in the future a desire arises to modify the quorum, we strongly recommend growing the size of membership rather than reducing the quorum threshold. The quorum exists to mitigate against bad actors, and is not implemented to prove a majority.
+- To reduce the feasibility of coercion or collusion, we recommend that members of the quorum be selected from trusted individuals within the Rust Project, community, and Foundation staff/board. This allows us to spread trust and responsibility among socially and fiduciarily responsible parties.
+- There is value in the members of the quorum generally residing in geographically and politically distributed locations most of the time. This RFC does not place any requirements regarding quorum member colocation for events or travel; the Rust Project and the quorum members can appropriately evaluate the value of such events and the corresponding risks.
+- Hardware keys shall be distributed for management and storage of signing keys by active members of the quorum. The Rust Foundation shall provide the hardware tokens at zero cost to the quorum members.
```

Comment:
if someone wanted to compromise this, swapping out all the hardware keys before sending them out to quorum members sounds like an ideal place, I think it would be better for quorum members to obtain hardware keys themselves and then the foundation can reimburse them (or members can just say it's a trivial cost so don't bother).

Reply:
@programmerjake The Foundation isn't going to purchase the keys and mail them out. The plan is for the Foundation to pay for the keys, not make the purchase. (This can happen in multiple different possible ways.)

Reply:
ok, i still think it's important that at least some of the hardware keys are obtained from a second source of some sort (even if it's just an independent distributor) so it isn't easy to e.g. have whatever mole in the supply chain send fake keys to the rust project since they were all purchased as a group or easily identifiable based off payment details.